onadata/apps/api/tests/viewsets/test_merged_xform_viewset.py

```diff
@@ -540,6 +552,46 @@ def test_get_osm_data_kwargs(self):
             'instance__xform_id': xform.pk
         })
 
+    # pylint: disable=invalid-name
+    def test_merged_with_attachment_endpoint(self):
+        merged_dataset = self._create_merged_dataset()
+        merged_xform = MergedXForm.objects.get(pk=merged_dataset['id'])
+        _make_submissions_merged_datasets(merged_xform)
+
+        # Attachment viewset works ok for filtered datasets
+        attachment_list_view = AttachmentViewSet.as_view({"get": "list"})
+        all_instances = Instance.objects.filter(xform__in=merged_xform.xforms.all())
+        for instance in all_instances:
+            _add_attachments_to_instances(instance)
+        request = self.factory.get(
+            "/?merged_xform=" + str(merged_xform.pk),
+            **self.extra)
+        response = attachment_list_view(request)
+        response_list = json.loads(json.dumps(response.data))
+        # We are removing all these because they keep changing
+        del response_list[0]['small_download_url']
+        del response_list[0]['medium_download_url']
+        del response_list[0]['download_url']
+        del response_list[0]['filename']
+        del response_list[0]['id']
+        del response_list[0]['instance']
+        del response_list[0]['url']
+        del response_list[0]['xform']
+        del response_list[1]['small_download_url']
+        del response_list[1]['medium_download_url']
+        del response_list[1]['download_url']
+        del response_list[1]['filename']
+        del response_list[1]['id']
+        del response_list[1]['instance']
+        del response_list[1]['url']
+        del response_list[1]['xform']
```

Comment:
```suggestion
        del response_list[0]['small_download_url']
        del response_list[0]['medium_download_url']
        del response_list[0]['download_url']
        del response_list[0]['filename']
        del response_list[0]['id']
        del response_list[0]['instance']
        del response_list[0]['url']
        del response_list[0]['xform']
        del response_list[1]['small_download_url']
        del response_list[1]['medium_download_url']
        del response_list[1]['download_url']
        del response_list[1]['filename']
        del response_list[1]['id']
        del response_list[1]['instance']
        del response_list[1]['url']
        del response_list[1]['xform']
```
Might just be nitpicking here but these seem like things that can be determined from the stored attachment object to me...

Reply:
I have refactored the test case, have a look!